### PR TITLE
update k8s version for testing and go to 1.20.8

### PR DESCRIPTION
**Pull request by cpanato**
/kind feature
/kind failing-test

**What this PR does / why we need it**:

- update k8s version for testing and go to 1.20.8
/hold for all tests

/assign @richardcase 

Fixes: #1016 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
update k8s version for testing and go to 1.20.8
```

### CODE REVIEW

**Comment by cpanato** [comment]
humm, interesting, here it is passing https://github.com/kubernetes-sigs/cluster-api-provider-gcp/pull/1019

**Comment by richardcase** [comment]
For the panic we can add `ctrl.SetLogger(klog.Background())` at the beginning of `TestE2E`.

**Comment by cpanato** [comment]
> For the panic we can add `ctrl.SetLogger(klog.Background())` at the beginning of `TestE2E`.

which panic? 🤔

**Comment by richardcase** [comment]
> which panic? 🤔

Actually its not a panic, just an error:

```
 [controller-runtime] log.SetLogger(...) was never called, logs will not be displayed:
goroutine 82 [running]:
runtime/debug.Stack() 
```

Sorry for the panic.

**Comment by richardcase** [comment]
Great work @cpanato , thanks for sorting this so quick. How did you know it was a problem with the versions?

/lgtm

**Comment by cpanato** [comment]
> Great work @cpanato , thanks for sorting this so quick. How did you know it was a problem with the versions?
> 
> /lgtm

to be honest just the feeling 😅

**Comment by cpanato** [comment]
humm but looks like it is something else in the ci-artifacts job @richardcase will take a look closer

**Comment by cpanato** [comment]
found

```
p kubernetes_semver=v1.29.0-alpha.0.819+bad6b29b9b7a26 kubernetes_source_type=http kubernetes_load_additional_imgs=false kubernetes_deb_version=1.26.7-1.1 kubernetes_rpm_version=1.26.7 no_proxy= pip_conf_file= python_path= redhat_epel_rpm=https://dl.fedoraproject.org/pub/epel/epel-release-latest-7.noarch.rpm epel_rpm_gpg_key= reenable_public_repos=true remove_extra_repos=false systemd_prefix=/usr/lib/systemd sysusr_prefix=/usr sysusrlocal_prefix=/usr/local load_additional_components=false additional_registry_images=false additional_registry_images_list= additional_url_images=false additional_url_images_list= additional_executables=false additional_executables_list= additional_executables_destination_path= additional_s3=false build_target=virt amazon_ssm_agent_rpm= enable_containerd_audit= --extra-vars  --extra-vars  --scp-extra-args  -e ansible_ssh_private_key_file=/tmp/ansible-key1827483788 -i /tmp/packer-provisioner-ansible1342639147 /home/prow/go/src/sigs.k8s.io/image-builder/images/capi/ansible/node.yml
```

fixing

**Comment by cpanato** [comment]
slack thread: https://kubernetes.slack.com/archives/C01D1RFEN9G/p1695027806376809

File: test/e2e/data/ccm/gce-cloud-controller-manager.yaml [hunk 1]
```diff
@@ -0,0 +1,369 @@
...
+          - --cloud-provider=gce
+          - --use-service-account-credentials=true
+          - --bind-address=127.0.0.1
+          - --secure-port=10258
+          - --cluster-name=${CLUSTER_NAME}
+          - --cluster-cidr=192.168.0.0/16
+          - --allocate-node-cidrs=true
+          - --configure-cloud-routes=false
+          - --cidr-allocator-type=CloudAllocator
+          - --controllers=cloud-node,cloud-node-lifecycle,nodeipam,service
```
Comment (aojea) [inline test/e2e/data/ccm/gce-cloud-controller-manager.yaml:62]: check that the kcm no longer runs the nodeipam service

**Comment by cpanato** [comment]
/test pull-cluster-api-provider-gcp-conformance-ci-artifacts

more running to make sure that works

**Comment by cpanato** [comment]
passed: https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api-provider-gcp/1018/pull-cluster-api-provider-gcp-conformance-ci-artifacts/1716063797619200000

**Comment by cpanato** [comment]
success the ci-artifacts test, final re-test

/test pull-cluster-api-provider-gcp-conformance-ci-artifacts

**Comment by cpanato** [comment]
this should be ready now @richardcase the `pull-cluster-api-provider-gcp-conformance-ci-artifacts` is now running fine, run 4-5 times with success

/unhold

@richardcase PTAL
cc @aojea 
any improvements and other things i think we can do in a follow up

**Comment by aojea** [comment]
+1 to the best of my knowledge ;)

**Review by richardcase — COMMENTED** [review]
Great work @cpanato 🥇  And thats very interesting about **cloud.config**.

/lgtm

**Pull request merged**